src/ray/core_worker/reference_count.cc

```diff
@@ -166,6 +166,20 @@ void ReferenceCounter::AddOwnedObject(const ObjectID &object_id,
   }
 }
 
+void ReferenceCounter::RemoveOwnedObject(const ObjectID &object_id) {
+  auto it = object_id_refs_.find(object_id);
+  if (it == object_id_refs_.end()) {
+    RAY_LOG(WARNING) << "Tried to remove reference for nonexistent owned object "
```

Comment:
Now I'm wondering if it should be a logged warning + a no-op or if it should actually be a fatal error (i.e. a `RAY_CHECK()`). Basically, should removing a nonexistent reference be a no-op, as we already do for removing a nonexistent local reference, or should it be considered an invalid use of the API and should error?

Reply:
I opted to make it a fatal error since it's an invalid use of the API. I think that this deviation from the no-op semantics of removing a nonexistent local reference is ok, since the semantics of `RemoveLocalReference` are already a bit different, and there could some calling logic where it's removing a local reference that may or may not have already been removed, and the no-op semantics make the caller's logic cleaner. Meanwhile, I think that `RemoveOwnedObject` should always be explicitly paired with an `AddOwnedObject` in the caller's logic, and not doing so should be considered an invalid use of the API.